### PR TITLE
[FEQ- 2056]] Nada/feat: add 404 component

### PR DESCRIPTION
**Pull request by nada-deriv**
Screenshot 2024-07-04 at 4 05 33 PM
Screenshot 2024-07-04 at 4 05 45 PM

https://github.com/deriv-com/p2p/assets/122768621/37547519-d261-4078-b1d8-7a4f49bf6396

### CODE REVIEW

File: src/assets/404.png
Comment (farrah-deriv) [inline src/assets/404.png:1]: can you please request this in quill?

File: src/components/Page404/Page404.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { useHistory } from 'react-router-dom';
+import { BUY_SELL_URL } from '@/constants';
+import { Localize } from '@deriv-com/translations';
+import { Button, Text, useDevice } from '@deriv-com/ui';
+import Icon404 from '../../assets/404.png';
```
Comment (ameerul-deriv) [inline src/components/Page404/Page404.tsx:5]: nit
```suggestion
import Icon404 from '@/assets/404.png';
```

File: src/components/Page404/Page404.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const Page404 = () => {
+    const { isDesktop } = useDevice();
+    const history = useHistory();
+    return (
+        <div className='page-404'>
+            <div>
+                <img alt='Page not found' src={Icon404} />
+            </div>
+            <div className='page-404__text'>
+                <Text size={isDesktop ? '2xl' : '2xl'} weight='bold'>
```
Comment (ameerul-deriv) [inline src/components/Page404/Page404.tsx:17]: wait is this supposed to be a different sizing or is it just 2xl here

Comment (nada-deriv) [reply src/components/Page404/Page404.tsx:17]: oh.. yeah. its supposed to be different 👍

File: src/routes/Router.tsx [hunk 3]
```diff
@@ -26,10 +27,10 @@ type TRoutesWithSubRoutes = {
...
     return (
         <Switch>
-            {route.routes?.map(subRoute => (
-                <Route key={subRoute.path} path={subRoute.path} render={props => <subRoute.component {...props} />} />
-            ))}
-            <Route path={route.path} render={props => <route.component {...props} />} />
+            {/* Render subroutes recursively with any number of nested routes */}
+            {route.routes?.map(subRoute => <RouteWithSubRoutes key={subRoute.path} {...subRoute} />)}
+            <Route component={route.component} exact path={route.path} />
+            <Route component={Page404} />
```
Comment (ameerul-deriv) [inline src/routes/Router.tsx:33]: do we need change this? wont it by default if theres no match for the routes, fall to the last Route below where the Page404 is?

Comment (nada-deriv) [reply src/routes/Router.tsx:33]: no.. it doesn't.. for the other change it goes to 404 .. but that doesn't handle nested routes.. so we need this for the nested ones..

Comment (ameerul-deriv) [reply src/routes/Router.tsx:33]: weird, it should fallback by default, but alright 👍 so all the nested routes are ok? did you check if it affects those routes like completing orders, share my ads

Comment (nada-deriv) [reply src/routes/Router.tsx:33]: yes. those are working fine.

**Review by ameerul-deriv — APPROVED** [review]

**Review by farrah-deriv — APPROVED** [review]

**Pull request merged**